lib/client.js

```diff
@@ -825,8 +825,9 @@ Request.prototype.end = function(fn){
 
   // set header fields
   for (var field in this.header) {
-    if (null == this.header[field]) continue;
-    xhr.setRequestHeader(field, this.header[field]);
+    if ({}.hasOwnProperty.call(this.header, field)) {
```

Comment:
Would it be marginally more efficient to do `Object.prototype.hasOwnProperty` here instead of declaring a new object literal each turn of the for loop?